File: service/test_jira.py
print(">>> Starting Jira test script...")

import os
from service.jira_service import (
    get_backlog_issues,
    create_issue,
    update_issue,
    transition_issue,
)

# Show environment variables (loaded by jira_service)
print(">>> Checking environment variables...")
print("JIRA_URL:", os.getenv("JIRA_URL"))
print("JIRA_EMAIL:", os.getenv("JIRA_EMAIL"))
print("JIRA_API_TOKEN:", "SET" if os.getenv("JIRA_API_TOKEN") else "MISSING")
print("JIRA_PROJECT:", os.getenv("JIRA_PROJECT") or "MISSING")

try:
    print(">>> Fetching issues...")
    issues = get_backlog_issues()
    print(f"Found {len(issues)} issues assigned to you.")
    for i in issues[:5]:
        print(i.key, "-", i.fields.summary)
except Exception as e:
    print("Error fetching issues:", e)

try:
    project_key = os.getenv("JIRA_PROJECT")
    if not project_key:
        print(">>> Skipping issue creation: JIRA_PROJECT env var not set.")
        print(">>> Set JIRA_PROJECT to your Jira project key (e.g., 'AIDE') to enable create/update/transition steps.")
    else:
        print(">>> Creating test issue...")
        new_issue = create_issue(project_key, "Test issue from NaiskosDev", "Created via API", issue_type="Task")
        print("Created issue:", new_issue.key)

        print(">>> Updating issue...")
        updated = update_issue(new_issue.key, {"summary": "Updated summary from NaiskosDev"})
        print("Updated issue:", updated.key)

        print(">>> Transitioning issue...")
        msg = transition_issue(new_issue.key, "In Progress")
        print(msg)
except Exception as e:
    print("Error creating/updating/transitioning issue:", e)

print(">>> Script finished.")

File: service/testJira.js
require('dotenv').config({ path: require('path').join(__dirname, '.env') });

const {
  getBacklogIssues,
  createIssue,
  updateIssue,
  transitionIssue
} = require('./jira_service');

(async () => {
  console.log(">>> Starting Jira test script...");

  console.log(">>> Checking environment variables...");
  console.log("JIRA_URL:", process.env.JIRA_URL);
  console.log("JIRA_EMAIL:", process.env.JIRA_EMAIL);
  console.log("JIRA_API_TOKEN:", process.env.JIRA_API_TOKEN ? "SET" : "MISSING");
  console.log("JIRA_PROJECT:", process.env.JIRA_PROJECT || "MISSING");

  try {
    console.log(">>> Fetching issues...");
    const issues = await getBacklogIssues();
    console.log(`Found ${issues.length} issues assigned to you.`);
    issues.slice(0, 5).forEach(i => {
      console.log(`${i.key} - ${i.fields.summary}`);
    });
  } catch (e) {
    console.error("Error fetching issues:", e.message);
  }

  try {
    const projectKey = process.env.JIRA_PROJECT;
    if (!projectKey) {
      console.log(">>> Skipping issue creation: JIRA_PROJECT env var not set.");
    } else {
      console.log(">>> Creating test issue...");
      const newIssue = await createIssue({
        projectKey,
        summary: "Test issue from NaiskosDev",
        description: "Created via API",
        issueType: "Task"
      });
      console.log("Created issue:", newIssue.key);

      console.log(">>> Updating issue...");
      const updated = await updateIssue(newIssue.key, {
        summary: "Updated summary from NaiskosDev"
      });
      console.log("Updated issue:", updated.key);

      console.log(">>> Transitioning issue...");
      const msg = await transitionIssue(newIssue.key, "In Progress");
      console.log(msg);
    }
  } catch (e) {
    console.error("Error creating/updating/transitioning issue:", e.message);
  }

  console.log(">>> Script finished.");
})();
